### ISSUE TITLE
ArrayField field name

### ISSUE THREAD

**Issue by matus1802** [joepuzzo/informed#143]
Hello,
first, thank you for perfect library. Version 2 brings good improvements, but I found this problem with ArrayField:
ArrayField field name is not correctly passed to children, where i.e. instead of myFieldName[] is always passed value field[]. I think, that the problem is with this line:
`field[${i}]`
where I would expect something like:
`${field}[${i}]`
Could you please have a look at it?
Thank you
Matus

**Comment by matus1802** [comment]
The other problem, which I have with this version (2.0.1) is, that the components are not rerendered when the state has changed. For example:
...
`
state = {
    loading:true
}
`
....after some API call....
`this.setState({loading: false})`
...
```
render() {
   return (
      <Form initialValues={this.defaultDef} getApi={(formApi) => this.setFormApi(formApi)} >
        {({formApi}) => (
                <Text id="name" field="name"
                    type={'text'}
                    validate={this.validators.name}
                    validateOnChange
                    validateOnBlur
                    disabled={this.state.loading}
                  />
     </Form>
)}
```

The Text is still disabled, even if the state.loading is false. In the previous version, it worked perfectly.
Thanks

**Comment by joepuzzo** [comment]
Ok so i think this is easy fix. I will look into it.

**Comment by joepuzzo** [comment]
And yup the array thing is totally a miss on my part!

**Comment by joepuzzo** [comment]
Both of these issues are now fixed! V 2.0.2

**Comment by matus1802** [comment]
thanks, works correctly ;)
